src/shared/components/Posts/hooks/usePosts.ts

```diff
@@ -0,0 +1,32 @@
+import { useQuery } from 'react-query';
+//interfaces
+import { IPostResponse } from '../interfaces/IPostsResponse';
+//services
+import { postsApiService } from '../services/postsApiService';
+
+interface IUsePosts {
+    postsData: IPostResponse[];
+    arePostsLoading: boolean;
+}
+export const usePosts = (): IUsePosts => {
+    const { data: postsData, isLoading: arePostsLoading } = useQuery(
+        ['posts'],
+        postsApiService.getAllPosts,
```

Comment:
We could put this in api.ts or something like that and just call it here it would be more readable